cmake/CeleritasAddTest.cmake

```
@@ -308,12 +308,22 @@ function(celeritas_add_test SOURCE_FILE)
 
     # Create an executable and link libraries against it
     add_executable(${_TARGET} "${SOURCE_FILE}" ${PARSE_SOURCES})
-    target_link_libraries(${_TARGET}
-      Celeritas::Test Celeritas::Core
+    set_target_properties(${_TARGET} PROPERTIES
+      CUDA_SEPARABLE_COMPILATION ON
+      CUDA_RUNTIME_LIBRARY Shared
+    )
+
+    if (BUILD_SHARED_LIBS AND CELERITAS_USE_CUDA)
+      celeritas_target_link_libraries(${_TARGET} Celeritas::Core VecGeom::vecgeom)
+    else()
+      celeritas_target_link_libraries(${_TARGET} Celeritas::Core)
+    endif()
```

Do you prefer:
```suggestion
    if (CELERITAS_USE_VecGeom)
      celeritas_target_link_libraries(${_TARGET} Celeritas::Core VecGeom::vecgeom)
    else()
      celeritas_target_link_libraries(${_TARGET} Celeritas::Core)
    endif()
```
or
```suggestion
    celeritas_target_link_libraries(${_TARGET} Celeritas::Core)
    if (CELERITAS_USE_VecGeom)
      celeritas_target_link_libraries(${_TARGET} VecGeom::vecgeom)
    endif()
```

---

I think the latter: less duplication here is better IMHO.